pages/blog/the-reference-rabbit-hole.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "The Reference Rabbit Hole"
+date: 2021-11-15T12:00:00+01:00
+type: Engineering
+tags:
+  - Specification
+  - JSON Schema 
+cover: /img/posts/the-reference-rabbit-hole.webp
+authors:
+  - name: Jonas Lagoni
+    photo: /img/avatars/jonaslagoni.webp
+    link: https://github.com/jonaslagoni
+    byline: AsyncAPI Maintainer
+---
+
+So [Sergio](https://github.com/smoya) and I, went down a little bit of a rabbit hole the last couple of days when we were discussing [Fran's proposal to solve publish/subscribe confusion](https://github.com/asyncapi/spec/issues/618), and I thought I would share the journey.
+
+## What is a reference?
+Just to start with the basics, a reference is something in AsyncAPI we use to allow for reusability, so we don't need to define things twice. In AsyncAPI we refer to this as the [Reference Object](https://www.asyncapi.com/docs/specifications/v2.2.0#referenceObject). The reference can either be local within the same document, external in another file or remotely.
+
+## The Trigger
+During the discussion, Sergio brought up that Fran was actually using an illegal reference, as he, in one of the examples, using a Reference Object for a server, which was not allowed.  More specifically, it was this example where he references the `mosquitto` server:
+
+```yaml
+asyncapi: 3.0.0
+...
+servers:
+  mosquitto:
+    $ref: 'common.asyncapi.yaml#/components/servers/mosquitto'
+```
+
+Immediately I said, "wait .... It's not?!". Cause before this, I had always used `$ref` quite extensively in my AsyncAPI documents and specifically used a reference for servers. And I knew that the tooling had no problems with the `$ref` as long as it was a valid reference. 
+
+But Sergio was absolutely right, a second look into the specification, `servers` are defined using [Servers Object](https://www.asyncapi.com/docs/specifications/v2.2.0#serversObject), which are defined using a map of [Server Object](https://www.asyncapi.com/docs/specifications/v2.2.0#serverObject)s. **NOT** `Server Object | Reference Object` as I expected.
```

Comment:
Some suggestions that IMHO simplify the sentence. Feel free to take it or discard it 👍

```suggestion
But Sergio was absolutely right. By taking a second look into the specification, I noticed `servers` are defined using [Servers Object](https://www.asyncapi.com/docs/specifications/v2.2.0#serversObject), which are basically a map of [Server Object](https://www.asyncapi.com/docs/specifications/v2.2.0#serverObject)s. **NOT** `Server Object | Reference Object` as I expected.
```

Reply:
I don't think this change is needed :)